### DISCUSSION TITLE
How to prevent "optimized dependencies changed. reloading" messages when running `dev`

### DISCUSSION THREAD

**Discussion by jrutila** [nuxt/nuxt#27700] [Questions]
I have this problem in my CI pipeline.
First, the pipeline runs `npm ci` to install deps. Then, it starts `nuxt dev` to start the backend and then runs Cypress tests against that backend. Now, on the first loads I get into the output messages like:
```
ℹ ✨ new dependencies optimized: @vueuse/core                                                                                                                 
ℹ ✨ optimized dependencies changed. reloading 
```

This causes the page to reload in the middle of the cypress run and fail the tests. Is there any option to somehow disable or force the optimization of the deps in this case? I can live with the dev environment reloading when I am developing but the tests can't cope with this.
There is related discussion in #26783, but it was closed with #27372. Having that change doesn't really help in this situation.
Reading here: https://vitejs.dev/config/dep-optimization-options, I have added the deps into the `include` array and that helps, but is there some other way to totally ignore that kind of behavior?

**Comment by TheAlexLichter** [comment]
Are you up on the latest version (also of the vueuse module)?
Could you provide a reproduction?

**Reply by jrutila** [reply]
I have the latest versions. This happens with many other dependencies, too. Here is my current workaround.
```
  vite: {
    optimizeDeps: {
      include: [
        "yup",
        "js-yaml",
        "txml",
        "@zip.js/zip.js",
        "@vueuse/core",
        "markdown-it",
        "markdown-it-footnote",
        "markdown-it-sup",
        "markdown-it-when",
        "markdown-it-video",
        "markdown-it-plain-text",
        "octokit",
      ],
    },
  },
```

**Reply by jrutila** [reply]
Nuxt project info:                                                                                                                                              

------------------------------
- Operating System: Windows_NT
- Node Version:     v18.18.0
- Nuxt Version:     3.12.2
- CLI Version:      3.12.0
- Nitro Version:    2.9.6
- Package Manager:  npm@10.7.0
- Builder:          -
- User Config:      ssr, runtimeConfig, app, extends, uiPro, modules, storyblok, imports, pinia, piniaPersistedstate, i18n, typescript, vue, hooks, ui, gtm, experimental,
 devtools, routeRules, nitro, vite
- Runtime Modules:  @nuxt/ui@2.17.0, @pinia/nuxt@0.5.1, @pinia-plugin-persistedstate/nuxt@1.2.0, @zadigetvoltaire/nuxt-gtm@0.0.13, @nuxtjs/robots@3.0.0, @nuxtjs/i18n@8.1.0, @nuxt/image@1.7.0, @storyblok/nuxt@6.0.10, @nuxt/eslint@0.3.13
- Build Modules:    -
------------------------------

**Comment by GlitchDevX** [comment]
I'm having the same issue with my playwright tests, I get an ECONNRESET whenever a dependency is optimized. This is causing my whole test to just "freeze", because it neither exits nor continues with anything. This only happens in my pipelines or on new clones.

Would it be possible to add a parameter to suppress this behavior like this:
```bash
npm run dev --suppress-dependency-optimization
```

It would make developing tests a whole lot easier for me and probably some other people.

Thanks for an answer in advance.

**Reply by GlitchDevX** [reply]
I was able to solve my issue with my tests by generating the static file and using npx serve to create a web server for testing. Would still be nice to get the parameter.

**Comment by zachnicoll** [comment]
This is issue is incredibly frustrating while developing - why would I want every page with a new dependency to reload by default?

Are we any closer to a solution?

**Reply by neilmerton** [reply]
I agree, I'm sure it's done for a reason, but it's not the best DX.

**Reply by jrutila** [reply]
Should we fire up a bug about this, as this is only a discussion. It is still happening with nuxt 3.16.0

**Reply by neilmerton** [reply]
Maybe, but part of a discussion is to understand. If there's a particular reason why this happens it would be great if it could be explained.

**Comment by stijn-q42** [comment]
This is quite an issue for me. It keeps overriding a virtually linked package, making local development a headache.

**Comment by vedmant** [comment]
I'm using `--suppress-dependency-optimization` but it has no effect. `"nuxt-dev": "nuxt dev --host=0.0.0.0 --qr=false --suppress-dependency-optimization",`

**Reply by GlitchDevX** [reply]
Because it's not a feature yet. I have just proposed that they could add this as a feature.

**Reply by jrutila** [reply]
Do you have a link to the proposal, so it could be followed?

**Comment by HendrikJan** [comment]
I have this problem too.
Could this setting in my `nuxt.config.ts` be the cause?:

```
ssr: false,
```

I cannot disable this setting to test it out, because my application totally depends on this setting and breaks in horrible ways when I disable it.
